chainer/optimizer.py

```diff
@@ -853,3 +853,58 @@ def __call__(self, rule, param):
         xp = cuda.get_array_module(grad)
         with cuda.get_device_from_array(grad):
             xp.clip(grad, self.lower_bound, self.upper_bound, out=grad)
+
+
+class GradientLARS(object):
+
+    """Optimizer/UpdateRule hook function for layer wise adaptive rate scaling.
+
+    See: `Large Batch Training of Convolutional Networks \
+          <https://arxiv.org/abs/1708.03888>`_
+
+    This hook function scales all gradient arrays to fit to the weight norm.
+
+    Args:
+        threashold (float): If weight norm is more than threshold,
+            this function scales all gradient arrays to fit weight norm.
+        weightdecay (float): Coefficient for the weight decay.
+        eps (float): Small value for the numerical stability.
+
+    Attributes:
+        threashold (float): If weight norm is more than threshold,
+            this function scales all gradient arrays to fit weight norm.
+        weightdecay (float): Coefficient for the weight decay.
+        eps (float): Small value for the numerical stability.
+    """
+    name = 'GradientLARS'
+    call_for_each_param = True
+
+    def __init__(self, threshold=1e-2, weightdecay=0.0, eps=1e-9):
```

Comment:
Could you let me know how did you determine the `eps` value. I could not find it in the papers (correct me if I miss it).

Reply:
I refer to source code in  https://github.com/bparr/lars written by tensorflow.

Reply:
OK, then please write in the doc string that your implementation is the version the paper (https://arxiv.org/abs/1801.03137) and introduced eps term, as described in section 6.1 of the paper.